CapstoneProject/Assets/Scripts/Core/Entity/Characters/NPC/NPCDialogue.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using Core.Gameplay;
+using Core.GameStates;
+using UnityEngine;
+
+public class NPCDialogue : MonoBehaviour, IInteractable
+{
+    bool triggered = false;
+    public void Interact()
+    {
+        if (!triggered)
+        {
+            Debug.Log("Interacted!");
+            GetComponent<DialogueManager>().StartInteraction();
```

Comment:
does each NPC have it's own dialogmanager? we might want a disalogmanager local reference instead of getting the component each interact. (check in start that it's not null, also maybe add a RequiresComponent DialogManager to the class itself if we are assuming it has one.